src/runtime/storage/pouchdb/pouch-db-storage.ts

```diff
@@ -96,6 +100,7 @@ export class PouchDbStorage extends StorageBase {
       return this.construct(id, type, key);
     } catch (err) {
       if (err.name && err.name === 'not_found') {
+        console.warn('connecting despite missing doc, returning null');
```

Comment:
Is it reasonable to return null here? Maybe throwing an error would make more sense?

It looks like Firebase equivalent should never return null, and throws errors in earlier sanity-checking of input params before ending up in `this.getInstance(key).storage;` and `storage` looks like it is expected to never be null.

Reply:
I've just restructured this so I think we'll be able to do this.

Right now the only place I see this happen is shell requesting the launcher arc.

Reply:
comment removed, only used in connectOrConstruct